packages/ui-notifications/src/use-notifications.tsx

```diff
@@ -41,20 +41,24 @@ export interface UseNotificationsInterface {
 export function useNotifications(): UseNotificationsInterface {
     const { push } = useContext(NotificationContext);
 
-    const notificationFromError = (status: Status): CurriedFunction2<string, any, void> =>
-        curry((title: string, err: AnyError): void => {
-            const message = err?.status ? `${err.status}: ${err.message}` : err.message;
-            push({
-                key: title,
-                message,
-                status,
-                title: `${status.toUpperCase()}: ${title}`,
+    const notificationFromError = React.useCallback(
+        (status: Status): CurriedFunction2<string, any, void> => {
+            return curry((title: string, err: AnyError): void => {
+                const message = err?.status ? `${err.status}: ${err.message}` : err.message;
+                push({
+                    key: title,
+                    message,
+                    status,
+                    title: `${Status.Error.toUpperCase()}: ${title}`,
+                });
             });
-        });
+        },
+        [push]
+    );
 
     return {
-        pushErrorNotification: notificationFromError(Status.ERROR),
+        pushErrorNotification: notificationFromError(Status.Error),
```

Comment:
Because the callback is executed here it would still run every re-render kind of defeating the memoization. You would need to have a useCallback defined that executes it and has this function as its deps called here.